src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -128,17 +130,27 @@ public void canSimulateApiUserFlow() throws URISyntaxException {
             apiUser.id,
             apiUser.email,
             TEMP_AUTH0_USER_PASSWORD);
-        HttpResponse<String> getTokenResponse = mockAuthenticatedPost(endpoint,
+        HttpResponse<String> getTokenResponse = mockAuthenticatedRequest(endpoint,
+            "",
             apiUser,
-            ""
+            HttpUtils.REQUEST_METHOD.POST,
+            false
         );
         LOG.info(getTokenResponse.body());
         assertEquals(HttpStatus.OK_200, getTokenResponse.statusCode());
+        TokenHolder tokenHolder = JsonUtils.getPOJOFromJSON(getTokenResponse.body(), TokenHolder.class);
+
+        // Define the header values to be used in requests from this point forward.
+        HashMap<String, String> headers = new HashMap<>();
+        headers.put("Authorization", "Bearer " + tokenHolder.getAccessToken());
+        headers.put("x-api-key", apiUser.apiKeys.get(0).value);
+
 
         // create an Otp user authenticating as an Api user.
-        HttpResponse<String> createUserResponse = mockAuthenticatedPost("api/secure/user",
-            apiUser,
-            JsonUtils.toJson(otpUser)
+        HttpResponse<String> createUserResponse = authenticatedRequest("api/secure/user",
+            JsonUtils.toJson(otpUser),
+            headers,
+            HttpUtils.REQUEST_METHOD.POST
```

Comment:
It's not entirely clear to me why there is a mix of authenticated and mocked requests.

Reply:
Authenticated requests are for API users once an access token has been acquired. The mock requests are for OTP users which don't have an access token as they were created via a third party. These requests are required to confirm unauthorization.